src/eager/mod.rs

```diff
@@ -0,0 +1,835 @@
+            return Err(status);
+        }
+        if self.data_type() != T::data_type() {
+            let msg = format!(
+                "The expected data type ({}) and underlying data type ({}) did not match.",
+                T::data_type(),
+                self.data_type()
+            );
+
+            status.set_lossy(crate::Code::InvalidArgument, &msg);
+            return Err(status);
+        }
+
+        // Safely unwrap since data_type was checked beforehand
+        unsafe { Ok(Tensor::from_tf_tensor(tf_tensor).unwrap()) }
+    }
+
+    ///
+    pub fn debug_info(&self) -> Result<DebugInfo> {
+        let status = Status::new();
+        let inner = unsafe { tf::TFE_TensorHandleTensorDebugInfo(self.inner, status.inner) };
+        if !status.is_ok() {
+            Err(status)
+        } else {
+            Ok(DebugInfo { inner })
+        }
+    }
+
+    ///
+    pub(crate) unsafe fn from_tensor_handle(
+        _ctx: &'a Context,
+        h: *mut tf::TFE_TensorHandle,
+    ) -> Self {
+        Self {
+            inner: h,
+            ctx: PhantomData,
+        }
+    }
+}
+
+/// Debugging/Profiling information for TFE_TensorHandle
+///
+/// TFE_TensorDebugInfo contains information useful for debugging and
+/// profiling tensors.
+#[derive(Debug)]
+pub struct DebugInfo {
+    inner: *mut tf::TFE_TensorDebugInfo,
+}
+impl_drop!(DebugInfo, TFE_DeleteTensorDebugInfo);
+
+impl DebugInfo {
+    /// Returns the number of dimensions used to represent the tensor on its device.
+    /// The number of dimensions used to represent the tensor on device can be
+    /// different from the number returned by TFE_TensorHandleNumDims.
+    /// The return value was current at the time of TFE_TensorDebugInfo creation.
+    pub fn on_device_num_dims(&self) -> i32 {
+        unsafe { tf::TFE_TensorDebugInfoOnDeviceNumDims(self.inner) }
+    }
+
+    /// Returns the number of elements in dimension `dim_index`.
+    /// Tensor representation on device can be transposed from its representation
+    /// on host. The data contained in dimension `dim_index` on device
+    /// can correspond to the data contained in another dimension in on-host
+    /// representation. The dimensions are indexed using the standard TensorFlow
+    /// major-to-minor order (slowest varying dimension first),
+    /// not the XLA's minor-to-major order.
+    /// On-device dimensions can be padded. TFE_TensorDebugInfoOnDeviceDim returns
+    /// the number of elements in a dimension after padding.
+    /// The return value was current at the time of TFE_TensorDebugInfo creation.
+    pub fn on_device_dim(&self, dim_index: i32) -> i64 {
+        unsafe { tf::TFE_TensorDebugInfoOnDeviceDim(self.inner, dim_index) }
+    }
+}
+
+struct Op {
+    inner: *mut tf::TFE_Op,
+}
+
+impl Op {
+    fn new(ctx: &Context, op_or_function_name: &str) -> Result<Op> {
+        let status = Status::new();
+
+        let op_or_function_name = CString::new(op_or_function_name).unwrap();
```

Comment:
These should avoid `unwrap()` in general.

Reply:
In order to avoid using unwrap, let's see if we can use CStr as the argument of new.